test/sparse/sparse.jl

```diff
@@ -1806,134 +1698,3 @@ end
-    B = convert(SparseMatrixCSC{Float32,Int32}, sprand(N, 1, 0.3))
-    # use different types to check internal type stability via allocation tests below
-    X = sparse(Array(A) .+ Array(B))
-    broadcast!(+, copy(X), A, B) # warmup for @allocated
-    @test (@allocated broadcast!(+, X, A, B)) == 0
-    X = sparse(Array(A) .* Array(B))
-    broadcast!(*, copy(X), A, B) # warmup for @allocated
-    @test (@allocated broadcast!(*, X, A, B)) == 0
-    X = sparse(broadcast(f, Array(A), Array(B)))
-    broadcast!(f, copy(X), A, B) # warmup for @allocated
-    @test (@allocated broadcast!(f, X, A, B)) == 0
-    # test broadcast! implementation for an arbitrary number of (input) sparse matrices
-    f(x, y, z) = x + y + z + 1
-    A = sprand(N, M, 0.2)
-    B = sprand(N, 1, 0.2)
-    C = convert(SparseMatrixCSC{Float32,Int32}, sprand(1, M, 0.2))
-    # use different types to check internal type stability via allocation tests below
-    X = sparse(Array(A) .+ Array(B) .+ Array(C))
-    broadcast!(+, copy(X), A, B, C) # warmup for @allocated
-    @test (@allocated broadcast!(+, X, A, B, C)) == 0
-    X = sparse(Array(A) .* Array(B) .* Array(C))
-    broadcast!(*, copy(X), A, B, C) # warmup for @allocated
-    @test (@allocated broadcast!(*, X, A, B, C)) == 0
-    X = sparse(broadcast(f, Array(A), Array(B), Array(C)))
-    broadcast!(f, copy(X), A, B, C) # warmup for @allocated
-    @test_broken (@allocated broadcast!(f, X, A, B, C)) == 0
-    # this last test allocates 16 bytes in the entry point for broadcast!, but none of the
-    # earlier tests of the same code path allocate. no allocation shows up with
-    # --track-allocation=user. allocation shows up on the first line of the entry point
-    # for broadcast! with --track-allocation=all, but that first line almost certainly
-    # should not allocate. so not certain what's going on.
-end
-
-@testset "test basic correctness of broadcast/broadcast! implementation for more than two (input) sparse matrices" begin
-    N, M = 10, 12
-    f(xs...) = sum(xs) + 1
-    A = sprand(N, M, 0.2)
-    B = sprand(N, 1, 0.2)
-    C = sprand(1, M, 0.2)
-    D = sprand(1, 1, 1.0)
-    E = spzeros(1, 1)
-    fA, fB, fC, fD, fE = map(Array, (A, B, C, D, E))
-    fX, fY = ones(fA), ones(fB)
-    X, Y = sparse(fX), sparse(fY)
-    for op in (+, *, f)
-        # horizontal expansion only
-        @test broadcast(op, A, B, B) == sparse(broadcast(op, fA, fB, fB))
-        @test broadcast!(op, X, A, B, B) == sparse(broadcast!(op, fX, fA, fB, fB))
-        # vertical expansion only
-        @test broadcast(op, B, D, E) == sparse(broadcast(op, fB, fD, fE))
-        @test broadcast!(op, Y, B, D, E) == sparse(broadcast!(op, fY, fB, fD, fE))
-        # separate horizontal and vertical expansion
-        @test broadcast(op, A, B, C) == sparse(broadcast(op, fA, fB, fC))
-        @test broadcast!(op, X, A, B, C) == sparse(broadcast!(op, fX, fA, fB, fC))
-        # simultaneous horizontal and vertical expansion
```

Comment:
Three input arguments suffice to test the relevant code paths, and indeed the testset titled `"broadcast[!] implementation capable of handling >2 (input) sparse vectors/matrices"` covers the relevant code. Thanks!